src/java/org/apache/cassandra/cql3/conditions/ColumnCondition.java

```diff
@@ -648,10 +648,16 @@ private ByteBuffer rowValue(Row row)
                 return cell == null ? null : cell.buffer();
             }
 
-            Cell<?> cell = getCell(row, column);
+            // getCell returns Cell<?>, which requires a method call to properly convert.
+            return getCellBuffer(getCell(row, column), userType);
+        }
+
+        private <V> ByteBuffer getCellBuffer(Cell<V> cell, UserType userType)
+        {
             return cell == null
                       ? null
-                      : userType.split(cell.buffer())[userType.fieldPosition(field)];
+                      : ByteBufferAccessor.instance.convert(userType.split(cell.accessor(), cell.value())[userType.fieldPosition(field)],
+                                                            cell.accessor());
```

Comment:
This change (and the smaller related ones at the `split()` call-sites) is just a consequence of having to generalize `TupleType#split()` to support `asComparableBytes()`, correct? This conversion is still a no-op as long as we have a buffer cell?